src/fabric/src/fabric2_server.erl

```diff
@@ -81,6 +84,14 @@ init(_) ->
     end,
     application:set_env(fabric, db, Db),
 
+    Dir = case config:get("fabric", "fdb_directory") of
+        undefined ->
+            [?DEFAULT_FDB_DIRECTORY];
+        Val when is_list(Val), length(Val) > 0 ->
```

Comment:
You probably want this as the first clause and the `undefined` clause as a `_` catch all. Otherwise if someone sets `fdb_directory = ` in the config the node won't be able to boot. Sorry I missed this earlier.

Reply:
Agreed

Reply:
It is a good practice to avoid using `length` (because `length` has O(n) complexity). Especially in cases where `Val <> []` would do.

Reply:
We look this up once per node start and then it gets saved to the application environment. So even if it is a large-ish value we'd only parse it once.